### ISSUE TITLE
Received unexpected out-of-order change - not in skippedSeqs

### ISSUE THREAD

**Issue by olivierboudet** [couchbase/sync_gateway#3787]
We are synchronizing changes from a pouchDB database to the sync gateway 2.0. Sometimes, the changes feed seems to be incorrect, changes are not in the correct order. We see in logs that some errors happen : "Received unexpected out-of-order change - not in skippedSeqs".

We have two instances of Sync Gateway 2.0 without bucket shadowing, behind a Nginx instance.
We have a three nodes Couchbase 4.6.3 cluster.

### Sync Gateway version
2.0.0(832;2d8a6c0)

### Operating system
Kubernetes 1.9 on Ubuntu Linux.

### Config file
```javascript
{
	"adminInterface":"0.0.0.0:4985",
	"interface":"0.0.0.0:4984",
	"log": [
		"*"
	],
	"databases": {
		"bucket1": {
			"unsupported": {
				"user_views": {
					"enabled": true
				}
			},
			"server": "http://couchbase-server:8091",
			"allow_empty_password": false,
			"bucket": "bucket1",
            "username":"%USER%",
			"password":"%PASSWORD%",
			"users": {
				"GUEST": {
					"disabled": true,
					"admin_channels": [
						"*"
					],
					"all_channels": [
						"*"
					]
				}
			}
		},
		{...}
	},
	"CORS": {
		"Origin": [
			"http://localhost:8100", "http://localhost:8080"
		],
		"LoginOrigin": [
			"http://localhost:8100", "http://localhost:8080"
		],
		"Headers": [
			"Content-Type",
			"Authorization"
		],
		"MaxAge": 1728000
	}
}

```

### Log output
https://gist.github.com/olivierboudet/39c75621b6eee822f70f86a99791f867

### Expected behavior
New document updates should have higher sequence number

### Actual behavior
There is errors "Received unexpected out-of-order change - not in skippedSeqs" in logs, and the documents update are not saved with the correct sequence order.

**Comment by adamcfraser** [comment]
I don't know of any scenarios that would result in that log message, under normal conditions.  I'd only expect that to be logged for sequences that were skipped, abandoned, and then subsequently arrived over the server's mutation feed.  By default Sync Gateway will wait for an hour before abandoning sequences, and will also then issue a final query-based check to see whether the sequence exists.  All of this makes it fairly unlikely that the sequence would then subsequently be associated with a document.

Can you provide a few more specifics on how you're testing/hitting this?  Specifically:

- Are there any processes potentially writing sequences to the bucket other than Sync Gateway (e.g. backup/restore, xdcr)?
- Can you provide more details on what's meant by "the changes feed seems to be incorrect, changes are not in the correct order".  Sample changes feed output would be ideal.
- The attached log excerpt doesn't provide any insight on how the unexpected sequence (e.g. 48739) was originally written.  Are full logs, including the original write, available?
-  Can you provide the sync gateway metadata (i.e. doc._sync) from one of the documents in question (e.g. doc `6113`)?

**Comment by olivierboudet** [comment]
@adamcfraser thanks, please find below more details for each of your questions.

- no process other than Sync Gateway writes to the bucket. There was no backup or restore process at this time. But we have upgraded a node to 5.1.1 recently (at 4th October), so our couchbase cluster is in compatibility mode (from 4.6.3) and you have rebalanced our cluster at 4th October.

- In our application, we call the changes endpoint with the since parameter (/bucket/_changes?active_only=true&style=all_docs&feed=normal&since=100). We process these changes, and save the last sequence returned by the endpoint. Later, we call again the endpoint with the parameter since updated with the previously saved last sequence. So, when the sequence 48739 is saved in the sync gateway, we have already processed changes after that one, and we are calling the sync gateway with a since parameter after the missed change. We are missing data in the way we process the changes feed.

- I found more logs searching for the sequence number 48739. There was some errors during around one hour : 

`2018-10-05T12:22:06.653Z Cache:   Deferring #48740 (1 now waiting for #48721...#48739)`
then some other errors (this is an extract of lines containing 48739 only) : 
```
2018-10-05T13:23:14.993Z WARNING: Skipped Sequence 48739 didn't show up in MaxChannelLogMissingWaitTime, and isn't available from the * channel view.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted. -- db.(*changeCache).CleanSkippedSequenceQueue.func1() at change_cache.go:245
2018-10-05T13:23:16.829Z Cache:   Querying 'channels' view for "*" (start=#48739, end=#48739, limit=0)
2018-10-05T13:23:16.839Z WARNING: Skipped Sequence 48739 didn't show up in MaxChannelLogMissingWaitTime, and isn't available from the * channel view.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted. -- db.(*changeCache).CleanSkippedSequenceQueue.func1() at change_cache.go:245
2018-10-06T07:36:23.797Z CRUD+: Saving doc (seq: #48739, id: 6113 rev: 26-af836a14c1e74c7ca4cd79fbe551ca86)
2018-10-06T07:36:23.800Z Cache:   Received unexpected out-of-order change - not in skippedSeqs (seq 48739, expecting 48981) doc "6113" / "26-af836a14c1e74c7ca4cd79fbe551ca86"
```

- This is the _sync metadata of one document (I have replaced channel names by * because it is confidential data, and removed the bodymaps):
```
 "_sync": {
    "rev": "31-7eae64c153a88f42d4bb05eea90a1a3e",
    "flags": 16,
    "sequence": 49069,
    "recent_sequences": [
      30515,
      30516,
      33088,
      33566,
      33567,
      39718,
      40169,
      40170,
      41136,
      48739,
      49018,
      49024,
      49052,
      49068,
      49069
    ],
    "history": {
      "revs": [
        "9-5fc56ef59e6f4e7cb7d3486a337b9a43",
        "19-a67e06e8a62d4d50951de5a2a98df484",
        "31-7eae64c153a88f42d4bb05eea90a1a3e",
        "14-c1880f3af35bf972ba37caf57acd62c1",
        "24-9858f2d269fe4b2a664954b84f375b85",
        "26-af836a14c1e74c7ca4cd79fbe551ca86",
        "20-5e6bafebd99488eb208e7ab4bfe10dd2",
        "27-99f5f66f46b82052be2429a79ee8527e",
        "6-e0f34e5435ae8062c36d9f0a49495d53",
        "7-98c82c72635945cbb793f0d7e2ae4c8e",
        "22-1da46a589d5769d2f05e1b5712f41526",
        "21-c4dfb8df4f8147a0c126f53208349e6d",
        "4-b20cdd7a1051520f09ee7f06bc4383d3",
        "10-07313db46062813ddc9ee6fe2ef69f88",
        "25-1781d9e15ede41039ea1729b5cc1af43",
        "1-cbf93d947be9ba2f7704b78267e2ca27",
        "3-24a1832670c64ea897790b74a512b8e0",
        "28-e7331e379b05c62bd1b5d6ad223a84ac",
        "30-63d95ce34008cc203d6a736277b42c86",
        "17-f81e8e67763531908ed08eeb412342e2",
        "20-3fe61f871bd347f39f9b394b313e8fa3",
        "12-52342c4c16cd4d289c98078e3b5d1ef6",
        "15-3ed4893bce6e41fa8b9a4779604d8a21",
        "5-3031cc86d8cd46a89166a141b57644e3",
        "21-4e12841fb14479284eead23fa5ae2619",
        "18-44dc53108ba1fc4bf04c622defe8dafa",
        "20-253c4b005a0347c08b2bc174f1be8325",
        "11-c5cd5350af551cef431eab821f7f883f",
        "15-6b2e7ff870f8407ca3fc48f6f5026370",
        "13-8fe3d871f644d04e6d881dcfbaee6faf",
        "2-8ee0428948c0d036760bb9da777dd738",
        "21-59bc0dc0917eb6248c686be0e640134a",
        "8-5ed848d8cbd6ec691329ba4b353ec06a",
        "23-23fa7f946ba3945870f9de0bceccc0d4",
        "29-aeb31b92f2031957c4047766eb6851bd",
        "16-21e9e545ace0b475443c66df1179dccc",
        "16-a45866de0d06398718457c1e694a74aa"
      ],
      "parents": [
        32,
        25,
        18,
        29,
        33,
        14,
        1,
        5,
        23,
        8,
        11,
        6,
        16,
        0,
        4,
        -1,
        30,
        7,
        34,
        35,
        1,
        27,
        3,
        12,
        26,
        19,
        1,
        13,
        3,
        21,
        15,
        20,
        9,
        10,
        17,
        22,
        28
      ],
      "deleted": [
        24,
        31,
        36
      ],
      "bodymap": {
        "24": "{}",
        "31": "{}",
        "36": "{}"
      },
      "channels": [
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        null,
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ],
        [
          "*"
        ]
      ]
    },
    "channels": {
      "*": null
    },
    "cas": "",
    "time_saved": "2018-10-08T08:03:30.13785462Z"
  }
```

**Comment by adamcfraser** [comment]
The document itself looks fine, and doesn't show any unexpected sequence ordering in the recent sequence history.

These logs:
```
2018-10-05T13:23:14.993Z WARNING: Skipped Sequence 48739 didn't show up in MaxChannelLogMissingWaitTime, and isn't available from the * channel view.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted. -- db.(*changeCache).CleanSkippedSequenceQueue.func1() at change_cache.go:245
2018-10-05T13:23:16.829Z Cache:   Querying 'channels' view for "*" (start=#48739, end=#48739, limit=0)
2018-10-05T13:23:16.839Z WARNING: Skipped Sequence 48739 didn't show up in MaxChannelLogMissingWaitTime, and isn't available from the * channel view.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted. -- db.(*changeCache).CleanSkippedSequenceQueue.func1() at change_cache.go:245
```
indicate that sequence 48739 is being abandoned after an hour of waiting for it to arrive over the feed.  At that point in time a document with that sequence did not exist in the bucket (based on the view query).  

This log line:
```
2018-10-06T07:36:23.797Z CRUD+: Saving doc (seq: #48739, id: 6113 rev: 26-af836a14c1e74c7ca4cd79fbe551ca86)
```
indicates that a document with that sequence was written to the bucket.  Obviously this is unexpected - the 'missing' sequence shouldn't be allocated 18 hours after it was originally expected/abandoned.

A few possibilities come to mind:
- Are the logs above from a single SG node, or a mix of log entries from both SG nodes?  If it's a mix, are the wall clocks significantly different between the two nodes?
- the upgrade process to 5.1.1 caused rollback in the bucket, including potential rollback of the _sync:seq value

If your upgrade process is now complete, I would suggest restarting the SG nodes to see whether you continue to see problems.  If so, we can at least rule out upgrade-related issues.

**Comment by olivierboudet** [comment]
@adamcfraser the logs comes from two SG instances. I checked just now, all clocks are strictly identical

Perhaps this particular case was an upgrade-related issue but I can see one other example of a similar error before we start our upgrade.

```
2018-10-03T07:22:13.959Z Cache:   Deferring #45713 (1 now waiting for #45712...#45712)
2018-10-03T07:22:13.959Z Cache:   Deferring #45713 (1 now waiting for #45712...#45712)
2018-10-03T07:22:13.971Z Cache:   Deferring #45714 (2 now waiting for #45712...#45712)
2018-10-03T07:22:13.971Z Cache:   Deferring #45714 (2 now waiting for #45712...#45712)
2018-10-03T07:22:13.974Z Cache:   Deferring #45715 (3 now waiting for #45712...#45712)
2018-10-03T07:22:13.974Z Cache:   Deferring #45715 (3 now waiting for #45712...#45712)
2018-10-03T07:22:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:22:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:22:34.395Z Sequences: get oldest, returning: 45712
2018-10-03T07:22:34.645Z Sequences: get oldest, returning: 45712
2018-10-03T07:22:37.456Z Sequences: get oldest, returning: 45712
2018-10-03T07:22:37.721Z Sequences: get oldest, returning: 45712
2018-10-03T07:23:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:23:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:23:07.357Z Sequences: get oldest, returning: 45712
2018-10-03T07:23:09.447Z Sequences: get oldest, returning: 45712
2018-10-03T07:23:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:23:30.016Z Sequences: get oldest, returning: 45712
2018-10-03T07:24:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:24:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:24:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T07:24:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:25:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:25:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:25:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:25:30.016Z Sequences: get oldest, returning: 45712
2018-10-03T07:26:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:26:00.019Z Sequences: get oldest, returning: 45712
2018-10-03T07:26:30.008Z Sequences: get oldest, returning: 45712
2018-10-03T07:26:30.016Z Sequences: get oldest, returning: 45712
2018-10-03T07:27:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:27:00.021Z Sequences: get oldest, returning: 45712
2018-10-03T07:27:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:27:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:28:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:28:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:28:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:28:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:29:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:29:00.019Z Sequences: get oldest, returning: 45712
2018-10-03T07:29:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T07:29:30.016Z Sequences: get oldest, returning: 45712
2018-10-03T07:30:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:30:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:30:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:30:30.019Z Sequences: get oldest, returning: 45712
2018-10-03T07:31:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:31:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:31:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T07:31:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:32:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:32:00.015Z Sequences: get oldest, returning: 45712
2018-10-03T07:32:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:32:30.018Z Sequences: get oldest, returning: 45712
2018-10-03T07:33:00.008Z Sequences: get oldest, returning: 45712
2018-10-03T07:33:00.015Z Sequences: get oldest, returning: 45712
2018-10-03T07:33:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:34:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:34:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:35:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:35:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:36:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:36:30.015Z Sequences: get oldest, returning: 45712
2018-10-03T07:37:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:37:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:38:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:38:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:39:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:39:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:40:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:40:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:41:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:41:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:42:00.017Z Sequences: get oldest, returning: 45712
2018-10-03T07:42:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:43:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:43:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:44:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:44:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:45:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:45:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:46:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:46:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:47:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:47:30.016Z Sequences: get oldest, returning: 45712
2018-10-03T07:48:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:48:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:49:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:49:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:50:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:50:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:51:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:51:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:52:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:52:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:53:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T07:53:22.918Z Sequences: get oldest, returning: 45712
2018-10-03T07:53:23.634Z Sequences: get oldest, returning: 45712
2018-10-03T07:53:30.015Z Sequences: get oldest, returning: 45712
2018-10-03T07:54:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T07:54:01.770Z Sequences: get oldest, returning: 45712
2018-10-03T07:54:02.491Z Sequences: get oldest, returning: 45712
2018-10-03T07:54:30.015Z Sequences: get oldest, returning: 45712
2018-10-03T07:55:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:55:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T07:56:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:56:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T07:57:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:57:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T07:58:30.038Z Sequences: get oldest, returning: 45712
2018-10-03T07:59:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T08:00:00.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:00:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:01:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:01:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:02:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:02:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:03:00.014Z Sequences: get oldest, returning: 45712
2018-10-03T08:03:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T08:04:00.018Z Sequences: get oldest, returning: 45712
2018-10-03T08:04:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:05:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:05:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T08:06:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T08:06:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:07:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:07:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:08:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:08:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:09:00.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:09:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:10:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:10:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:11:00.012Z Sequences: get oldest, returning: 45712
2018-10-03T08:11:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T08:12:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:12:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:13:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:13:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:14:00.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:14:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:15:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:15:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:16:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:16:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:17:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:17:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:18:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:18:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:19:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:19:30.012Z Sequences: get oldest, returning: 45712
2018-10-03T08:20:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:20:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:21:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:21:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T08:22:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:22:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:23:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:23:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T08:24:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:24:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:25:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:25:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:26:00.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:26:16.766Z Sequences: get oldest, returning: 45712
2018-10-03T08:26:17.451Z Sequences: get oldest, returning: 45712
2018-10-03T08:26:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:26:57.433Z Sequences: get oldest, returning: 45712
2018-10-03T08:26:58.151Z Sequences: get oldest, returning: 45712
2018-10-03T08:27:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:27:24.697Z Sequences: get oldest, returning: 45712
2018-10-03T08:27:26.478Z Sequences: get oldest, returning: 45712
2018-10-03T08:27:30.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:28:00.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:28:30.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:29:00.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:29:30.014Z Sequences: get oldest, returning: 45712
2018-10-03T08:30:00.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:30:30.010Z Sequences: get oldest, returning: 45712
2018-10-03T08:31:00.011Z Sequences: get oldest, returning: 45712
2018-10-03T08:31:30.013Z Sequences: get oldest, returning: 45712
2018-10-03T08:32:00.009Z Sequences: get oldest, returning: 45712
2018-10-03T08:32:11.176Z Cache:   Querying 'channels' view for "*" (start=#45712, end=#45712, limit=0)
2018-10-03T08:32:11.617Z WARNING: Skipped Sequence 45712 didn't show up in MaxChannelLogMissingWaitTime, and isn't available from the * channel view.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted. -- db.(*changeCache).CleanSkippedSequenceQueue.func1() at change_cache.go:222
2018-10-03T08:32:12.352Z Cache:   Querying 'channels' view for "*" (start=#45712, end=#45712, limit=0)
2018-10-03T08:32:12.368Z WARNING: Skipped Sequence 45712 didn't show up in MaxChannelLogMissingWaitTime, and isn't available from the * channel view.  If it's a valid sequence, it won't be replicated until Sync Gateway is restarted. -- db.(*changeCache).CleanSkippedSequenceQueue.func1() at change_cache.go:222
2018-10-03T08:32:30.009Z Cache:   Querying 'channels' view for "*" (start=#45712, end=#46772, limit=0)
2018-10-03T12:17:23.609Z CRUD+: Saving doc (seq: #45712, id: 8074 rev: 1-2a7b520706174b3a6a5c5f1c77936663)
2018-10-03T12:17:23.611Z Cache: Received #45712 after   2ms ("8074" / "1-2a7b520706174b3a6a5c5f1c77936663")
2018-10-03T12:17:23.611Z Cache:     #45712 ==> channel "*"
2018-10-03T12:17:23.611Z Cache:   Received unexpected out-of-order change - not in skippedSeqs (seq 45712, expecting 46822) doc "8074" / "1-2a7b520706174b3a6a5c5f1c77936663"
2018-10-03T12:17:23.611Z Cache: Not adding change #45712 ==> channel "*", since it will be immediately pruned
2018-10-03T12:17:23.611Z Cache: Not adding change #45712 ==> channel "*", since it will be immediately pruned
2018-10-03T12:17:23.611Z Cache: Received #45712 after   2ms ("8074" / "1-2a7b520706174b3a6a5c5f1c77936663")
2018-10-03T12:17:23.611Z Cache:     #45712 ==> channel "*"
2018-10-03T12:17:23.611Z Cache:   Received unexpected out-of-order change - not in skippedSeqs (seq 45712, expecting 46822) doc "8074" / "1-2a7b520706174b3a6a5c5f1c77936663"

```

It is not always reproducible. In the log history I have (15 days), I can see this error only on 3th October and 5th/6th October.
I do not receive any error since 6th October without restarting the SG instances.

**Comment by adamcfraser** [comment]
@olivierboudet If you see a recurrence of this issue, it will help to collect full Sync Gateway and Couchbase Server logs at the time.  Thanks in advance.

**Comment by olivierboudet** [comment]
@adamcfraser 
We have a new case of errors "Received unexpected out-of-order change - not in skippedSeqs". I have collected the logs of our instances. Can I send it to you privately to avoid exposing confidential data ?

**Comment by olivierboudet** [comment]
Hello @adamcfraser,

The error happen one more time today. I have attached the logs of the 2 Sync Gateway instances and the 3 Couchbase Server instances.

You can find all logs here : https://we.tl/t-UoFbMzL0Po
The issue occurs at 2018-10-22T07:08:19.853Z in sg_log.2.log

**Comment by olivierboudet** [comment]
For your information, since the last time, I have kept only one SG instance and the issue does not happened since (for the moment...).

**Comment by adamcfraser** [comment]
Tracking this issue w/ https://issues.couchbase.com/browse/CBG-253.

**Comment by bbrks** [comment]
Closing issue, as CBG-253 was released in Sync Gateway 2.6.0
